lizard_connector/connector.py

```diff
@@ -383,7 +402,57 @@ def upload(self, data, uuid=None):
             data (dict): Dictionary with the data to post to the api
         """
         if uuid:
-            post_url = urljoin(urljoin(self.base_url, uuid), 'data')
+            post_url = urljoin(
+                self.base_url, "{}/{}/".format(uuid, sub_endpoint))
         else:
             post_url = self.base_url
         return self.post(post_url, data)
+
+
+class Client(Connector):
+    __endpoints = ()
+
+    def __init__(self, base="https://demo.lizard.net", username=None,
+                 password=None, parser=DEFAULT_PARSER,
+                 version=DEFAULT_API_VERSION, parser_kwargs=None,
+                 **kwargs):
+        self.api_version = version
+        if username is not None:
+            kwargs.update({
+                "username": username,
+                "password": password or getpass.getpass()
+            })
+        self.base = base
+        for endpoint in self.endpoints:
+            endpoint_params = dict(
+                endpoint=endpoint.replace('_', '-'),
+                base=self.base,
+                version=self.api_version,
+                parser=parser,
+                parser_kwargs=parser_kwargs)
+            endpoint_params.update(kwargs)
+            setattr(
+                self, endpoint, Endpoint(**endpoint_params)
+            )
+            if endpoint in DETAIL_ENDPOINTS:
+                endpoint_params["detail"] = True
+                setattr(
+                    getattr(self, endpoint),
+                    "data",
+                    Endpoint(**endpoint_params)
+                )
+        super(Client, self).__init__(
+            parser=parser, parser_kwargs=parser_kwargs, **kwargs)
+
+    @property
+    def endpoints(self):
+        if not self.__endpoints:
```

Comment:
@cachedproperty is netter (oh, dat is een Django ding geloof ik -- kopieren?)

Reply:
ik wilde niet afhankelijk zijn van een dependency @cached property zet dit in de `__dict__`. Dat zal ik doen.